opengrok-indexer/src/test/java/org/opengrok/indexer/history/GitRepositoryTest.java

```diff
@@ -86,13 +97,133 @@ public void tearDown() {
         instance = null;
     }
 
+    private void checkCurrentVersion(File root, int timestamp, String commitId, String shortComment)
+            throws Exception {
+        GitRepository gitrepo = (GitRepository) RepositoryFactory.getRepository(root);
+        assertNotNull(gitrepo);
+        String ver = gitrepo.determineCurrentVersion();
+        assertNotNull(ver);
+        Date date = new Date((long) (timestamp) * 1000);
+        assertEquals(Repository.format(date) + " " + commitId + " " + shortComment, ver);
+    }
+
     @Test
     public void testDetermineCurrentVersion() throws Exception {
```

Comment:
I feel the tests are overly long and are testing multiple things at once. I'd personally break them into multiple tests, e.g.
- testDetermineCurrentVersionForSingleRepository
- testBuildTagListForSingleHeadTag
- testBuildTagListForMultipleHeadTags
- testBuildTagListForTagOnSpecificCommit

If such a test fails and is short, then I find it easier to determine what is actually wrong.

Reply:
make sense, refactored.